### PR TITLE
Add user controller

### LINKED ISSUE #89: Create controller for adding users

### PR DESCRIPTION
**Pull request by thu5**
This PR adds a controller for handling operations on users. It also adds utility functions to encrypt/decrypt refresh tokens, and to get the user ID from ID tokens. Closes #89.

### CODE REVIEW

File: server/config.src.yaml [hunk 1]
```diff
@@ -4,6 +4,7 @@ projectId: projectid
 databaseId: (default)
 locationId: us-central1
 appEngineService: rules-server
+refreshTokenKey: arefreshtokenkey
```
Comment (cathyang-work) [inline server/config.src.yaml]: The format should be consistent. and `a` can be ignored. `refresh-token-key?`

File: server/test/utils_test.dart [hunk 2]
```diff
@@ -63,4 +65,32 @@ void main() {
...
+
+      final encrypted = encryptRefreshToken(testRefreshToken, refreshTokenKey);
+      final decrypted = decryptRefreshToken(encrypted, refreshTokenKey);
+
+      expect(decrypted, equals(testRefreshToken));
+    });
+  });
+
+  group('Get user ID from ID token', () {
+    test('encryptRefreshToken() ', () async {
```
Comment (gg-qinp) [inline server/test/utils_test.dart]: getUserId()

File: server/lib/controller/user_controller.dart [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    ContentType('application', 'x-protobuf')
+  ];
+
+  /// Creates a user with a user ID and encrypted refresh token.
+  ///
+  /// Throws an [ArgumentError] if the request does not contain an Authorization
+  /// header. Throws an [ApiRequestError] if the Firestore API returns an error.
+  /// Returns a 200 OK [Response] upon success.
+  @Operation.post()
+  Future<Response> createUser(@Bind.body() List<int> body) async {
```
Comment (gg-qinp) [inline server/lib/controller/user_controller.dart:34]: nit:  In general the create() method needs to return the resources created in stead of empty() following the restful design . I assume this function will be called in the background of the server, so it is not customer initiated and visible operation. Also seems that directly binding the body with a message like User has better readability. Maybe we can add a TODO here, and refactor later.

**Review by gg-qinp — APPROVED** [review]

File: server/test/utils_test.dart (hunk 2 above)
Comment (thu5) [reply server/test/utils_test.dart]: 😅

File: server/lib/controller/user_controller.dart (hunk 3 above)
Comment (thu5) [reply server/lib/controller/user_controller.dart:34]: Ok, that makes sense

**Review by cathyang-work — APPROVED** [review]

**Pull request merged**